### PR TITLE
Fixed FormManager error (missing ServiceLocator in Factory) and PHPUnit Tests in PHPUnit 5.5.2

### PR DESCRIPTION
**Pull request by Fahl-Design**
- Fixed PHPUnit tests
-- getMock is deprecated
-- zend hydrator package was missing added to composer.json and fixed call

### CODE REVIEW

**Comment by bramstroker** [comment]
In which use case you got a problem with service locator missing?

Btw. you can use the `FormElementManager` now which is part of ZF2 itself. You can add you form aliases and factories to it and have exactly the same behaviour as with the `FormManager` in my module. I think we should mark the `FormManager` deprecated and trigger deprecated error so we can remove it in the near future.

About the PHP 5.6 version restriction. This is a major change and we can only do that when releasing a new major version (i.e. 1.0.0). I had been thinking about that and I'm positive about doing that now as most people have been migrated to this version or PHP 7. Other users can still use old stable packages in the 0.* range.

File: composer.json [hunk 1]
```diff
@@ -17,20 +17,13 @@
...
-        "zendframework/zend-http": "~2.3",
-        "zendframework/zend-json": "~2.3",
-        "zendframework/zend-mvc": "~2.3",
-        "zendframework/zend-i18n": "~2.3",
-        "zendframework/zend-inputfilter": "~2.3",
-        "zendframework/zend-servicemanager": "~2.3",
-        "zendframework/zend-validator": "~2.3",
-        "zendframework/zend-view": "~2.3"
+        "php": ">=5.6 || >= 7.0",
+        "zendframework/zendframework": "2.*"
```
Comment (bramstroker) [inline composer.json]: Please don't depend on the complete zendframework, but only on the seperate packages.

File: tests/StrokerFormTest/Renderer/JqueryValidate/RendererTest.php [hunk 2]
```diff
@@ -94,7 +94,7 @@ protected function createForm($alias, $emailFieldName = 'email')
         $factory = new Factory();
         $form = $factory->createForm(
             array(
-                'hydrator' => 'Zend\Stdlib\Hydrator\ArraySerializable',
+                'hydrator' => \Zend\Hydrator\ArraySerializable::class,
```
Comment (bramstroker) [inline tests/StrokerFormTest/Renderer/JqueryValidate/RendererTest.php]: Please import this class

File: tests/StrokerFormTest/Renderer/JqueryValidate/Rule/AbstractRuleTest.php [hunk 3]
```diff
@@ -65,7 +65,7 @@ protected function createElement()
     public function setUp()
     {
         $this->rule = $this->createRule();
-        $this->translatorMock = $this->getMock('Zend\I18n\Translator\Translator', array('translate'));
+        $this->translatorMock = $this->createMock('Zend\I18n\Translator\Translator');
```
Comment (bramstroker) [inline tests/StrokerFormTest/Renderer/JqueryValidate/Rule/AbstractRuleTest.php]: No we depend on PHP 5.6 please also use `::class` here and import the class.

**Comment by bramstroker** [comment]
Thanks for your efforts. Which tool did you use to change the code style? PSR fixer?

**Comment by Fahl-Design** [comment]
your welcome, i got your points and made some chaneges as you can see. Well mabe it time for a new release for PHP >= 5.6 at least to get this sweet module runing with it and the latest zf2 2 version and a newer phpunit version. 
I'm also using this branch for an zf2 project i'm working on and for code style i used phpstorm with an zend preset and an .editorconfig from an official zend module (but with 160 chars per line because WTF 80 chars per line?!)

**Comment by Fahl-Design** [comment]
I added an php-cs fixer with a config from zend/session and changed the .travis.yml to execute them and show errors (with fixes provided)

**Comment by bramstroker** [comment]
@Fahl-Design Could you please retarget this PR against the develop branch? Than I can merge soon.

**Pull request merged**